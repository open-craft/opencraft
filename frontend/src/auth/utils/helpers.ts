import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
