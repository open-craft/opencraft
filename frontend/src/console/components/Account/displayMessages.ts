const messages = {
  explanation: {
    defaultMessage:
      'There are many factors that affect the speed at which updates will ' +
      'reflect on your site. Please allow up to 2 hours for your site to be updated.',
    description: 'Brief explanation on deployment duration.'
  },
  account: {
    defaultMessage: 'Account details',
    description: ''
  },
  updatePassword: {
    defaultMessage: 'Update password',
    descriptin: ''
  },
  fullName: {
    defaultMessage: 'Your full name*',
    description: ''
  },
  fullNameHelp: {
    defaultMessage: 'Name and surname',
    description: ''
  },
  email: {
    defaultMessage: 'Email address*',
    description: ''
  },
  emailHelp: {
    defaultMessage:
      'This is also your account name, and where we will send important updates',
    description: ''
  },
  oldPassword: {
    defaultMessage: 'Current password*',
    description: ''
  },
  newPassword: {
    defaultMessage: 'New password*',
    description: ''
  },
  newPasswordHelp: {
    defaultMessage:
      'Password must have 8 characters, 1 number, 1 capital letter and 1 special character.',
    description: ''
  },
  saveAccountDetails: {
    defaultMessage: 'Save',
    description: ''
  },
  saveNewPassword: {
    defaultMessage: 'Save',
    description: ''
  }
};

export default messages;
