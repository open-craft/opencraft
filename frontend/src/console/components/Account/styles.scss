@import '~styles/theme';

.account-heading {
  font-family: PlayfairDisplay;
  font-size: 30px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #1e4c59;
  margin-bottom: 30px;
}

.form-group p {
  color: #7a7a7a !important;
  font-size: 14px !important;
}

.form-group .form-label {
  font-size: 16px !important;
  font-family: Chivo !important;
  color: #0f1f24;
  letter-spacing: 0.16px !important;
}

.console-page .console-page-container {
  background-color: rgba(109, 156, 174, 0.2);
  padding-top: 105px;
  padding-bottom: 172.5px;
}

.console-page-content .container-fluid {
  width: 90%;
}

.account-details, .password-change {
  padding-top: 63px;
  padding-bottom: 89px;
}

.password-change {
  margin-top: 51px;
}

.account-details, .password-change {
  box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.3);

  button {
    width: 120px;
    height: 49px;
    font-size: 14px;
    font-family: Chivo;
  }
}