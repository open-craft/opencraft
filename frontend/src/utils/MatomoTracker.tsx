import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
