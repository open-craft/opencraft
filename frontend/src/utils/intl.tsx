import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
