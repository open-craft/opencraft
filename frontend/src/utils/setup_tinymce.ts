import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
