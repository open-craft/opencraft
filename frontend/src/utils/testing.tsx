import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
