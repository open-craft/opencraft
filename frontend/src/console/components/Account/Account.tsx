import * as React from 'react';
import { connect } from 'react-redux';
import { RootState } from 'global/state';
import { Col, Row, Form, Button, Spinner } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';
import { InstancesModel } from 'console/models';
import { TextInputField } from 'ui/components';
import {
  updateAccountDetails,
  getAccountDetails,
  changePassword
} from 'console/actions';
import messages from './displayMessages';
import { ConsolePage } from '../ConsolePage';
import './styles.scss';

interface ActionProps {
  updateAccountDetails: Function;
  getAccountDetails: Function;
  changePassword: Function;
}

interface State {
  [key: string]: string | boolean;
  fullName: string;
  email: string;
  oldPassword: string;
  newPassword: string;
}

interface StateProps extends InstancesModel {}

interface Props extends StateProps, ActionProps {}

export class AccountComponent extends React.PureComponent<Props, State> {
  constructor(props: Props, state: State) {
    super(props);

    const { account } = this.props;

    this.state = {
      fullName: account.fullName,
      email: account.email,
      oldPassword: '',
      newPassword: ''
    };
  }

  public componentDidMount() {
    this.props.getAccountDetails();
  }

  public componentDidUpdate(prevProps: Props, prevState: State) {
    this.updateInitialState(prevState);
  }

  private updateInitialState = (prevState: State) => {
    const { account } = this.props;

    if (account && account.fullName && !prevState.fullName) {
      this.setState({
        fullName: account.fullName,
        email: account.email,
        oldPassword: '',
        newPassword: ''
      });
    }
  };

  private onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const field = e.target.name;
    const { value } = e.target;

    this.setState({
      [field]: value
    });
  };

  private saveAccountDetails = () => {
    this.props.updateAccountDetails({
      fullName: this.state.fullName,
      email: this.state.email
    });
  };

  private saveNewPassword = () => {
    this.props.changePassword({
      oldPassword: this.state.oldPassword,
      newPassword: this.state.newPassword
    });
  };

  public render() {
    return (
      <ConsolePage
        showSidebar={false}
        contentLoading={this.props.loading}
        showToolbar={false}
      >
        <Row className="justify-content-center bg-white account-details">
          <Col md={8}>
            <div className="account-heading">
              <p>
                <WrappedMessage id="account" messages={messages} />
              </p>
            </div>
            <Form>
              <TextInputField
                fieldName="fullName"
                value={this.state.fullName}
                onChange={this.onChange}
                messages={messages}
                type="username"
                autoComplete="username"
              />
              <TextInputField
                fieldName="email"
                value={this.state.email}
                onChange={this.onChange}
                messages={messages}
                type="email"
                autoComplete="email"
              />

              <Button
                variant="primary"
                onClick={() => {
                  this.saveAccountDetails();
                }}
              >
                {this.props.accountDetailsUpdating && (
                  <Spinner animation="border" size="sm" className="spinner" />
                )}
                <WrappedMessage messages={messages} id="saveAccountDetails" />
              </Button>
            </Form>
          </Col>
        </Row>

        <Row className="justify-content-center bg-white password-change">
          <Col md={8}>
            <div className="account-heading">
              <p>
                <WrappedMessage id="updatePassword" messages={messages} />
              </p>
            </div>
            <Form>
              <TextInputField
                fieldName="oldPassword"
                value={this.state.oldPassword}
                onChange={this.onChange}
                messages={messages}
                type="password"
                autoComplete="password"
              />
              <TextInputField
                fieldName="newPassword"
                value={this.state.newPassword}
                onChange={this.onChange}
                messages={messages}
                type="password"
                autoComplete="password"
              />

              <Button
                variant="primary"
                onClick={() => {
                  this.saveNewPassword();
                }}
              >
                {this.props.passwordUpdating && (
                  <Spinner animation="border" size="sm" className="spinner" />
                )}
                <WrappedMessage messages={messages} id="saveNewPassword" />
              </Button>
            </Form>
          </Col>
        </Row>
      </ConsolePage>
    );
  }
}

export const Account = connect<{}, ActionProps, {}, Props, RootState>(
  (state: RootState) => state.console,
  {
    updateAccountDetails,
    getAccountDetails,
    changePassword
  }
)(AccountComponent);
