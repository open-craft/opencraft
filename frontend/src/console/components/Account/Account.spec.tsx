import React from 'react';
import { setupComponentForTesting } from 'utils/testing';
import { Account } from './Account'

describe('Account tests', () => {
  it('Correctly renders page without data', () => {
    const tree = setupComponentForTesting(
      <Account />,
      {
        console: {
          loading: false,
          account: {
            fullName: '',
            email: '',
            oldPassword: '',
            newPasword: ''
          },
          activeInstance: {
            data: null
          }
        }
      }
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Correctly renders page with data', () => {
    const tree = setupComponentForTesting(
      <Account />,
      {
        console: {
          loading: false,
          account: {
            fullName: 'Anon Y. Mous',
            email: 'anon@ymous.com',
            oldPassword: '',
            newPasword: ''
          },
          activeInstance: {
            data: {
              id: 1,
              instanceName: "test",
              subdomain: "test",
              lmsUrl: "test-url",
              studioUrl: "test-url",
              isEmailVerified: true,
            },
            deployment: {
              status: "preparing",
              undeployedChanges: [],
              deployedChanges: null,
              type: 'admin',
            }
          },
          instances: [{
            id: 1,
            instanceName: "test",
            subdomain: "test",
          }]
        }
      }
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
