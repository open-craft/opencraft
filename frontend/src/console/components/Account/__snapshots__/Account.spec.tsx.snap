// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account tests Correctly renders page with data 1`] = `
<div
  className="console-page"
>
  <div
    className="console-page-container"
  >
    <div
      className="console-page-content row"
    >
      <div
        className="container-fluid"
      >
        <div
          className="customization-form"
        >
          <div
            className="loading"
          >
            <i
              className="fas fa-2x fa-sync-alt fa-spin"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Account tests Correctly renders page without data 1`] = `
<div
  className="console-page"
>
  <div
    className="console-page-container"
  >
    <div
      className="console-page-content row"
    >
      <div
        className="container-fluid"
      >
        <div
          className="customization-form"
        >
          <div
            className="loading"
          >
            <i
              className="fas fa-2x fa-sync-alt fa-spin"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
